Source/WebCore/dom/ScriptExecutionContext.cpp

```diff
@@ -390,6 +390,10 @@ void ScriptExecutionContext::stopActiveDOMObjects()
         return ShouldContinue::Yes;
     });
     m_deferredPromises.clear();
+
+    m_nativePromiseRequests.forEach([] (auto& request) {
+        request.disconnect();
```

Comment:
you need to test that the request hasn't been completed as it would otherwise assert.
```suggestion
        if (request)
            request.disconnect();
```

Reply:
If request was complete, request would have been destroyed.